packages/amplify_datastore_plugin_interface/lib/src/types/conflict_handler/datastore_conflict_handler.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+
+typedef DataStoreConflictHandler = ConflictResolutionDecision Function(
+    ConflictData);
+
+class ConflictData {
+  Model local;
+  Model remote;
+
+  ConflictData(this.local, this.remote);
+
+  ConflictData.fromJson(ModelType modelType, Map<String, dynamic> localJson,
+      Map<String, dynamic> remoteJson)
+      : local = modelType.fromJson(
+            new Map<String, dynamic>.from(localJson["serializedData"])),
+        remote = modelType.fromJson(
+            new Map<String, dynamic>.from(remoteJson["serializedData"]));
+}
+
+enum ResolutionStrategy { APPLY_REMOTE, RETRY_LOCAL, RETRY }
```

Comment:
I won't mandate it here since we haven't cleaned up older code which does this - but please use camelCase for enum types per Dart conventions